LaunchDarkly/LaunchDarkly/ServiceObjects/Cache/KeyedValueCache.swift

```diff
@@ -1,25 +1,14 @@
 import Foundation
-
+import OSLog
 // sourcery: autoMockable
-protocol KeyedValueCaching {
+public protocol KeyedValueCaching {
     func set(_ value: Data, forKey: String)
     func data(forKey: String) -> Data?
-    func dictionary(forKey: String) -> [String: Any]?
     func removeObject(forKey: String)
     func removeAll()
     func keys() -> [String]
 }
 
-extension UserDefaults: KeyedValueCaching {
-    func set(_ value: Data, forKey: String) {
-        set(value as Any?, forKey: forKey)
-    }
-
-    func removeAll() {
-        dictionaryRepresentation().keys.forEach { removeObject(forKey: $0) }
-    }
-
-    func keys() -> [String] {
-        dictionaryRepresentation().keys.map { String($0) }
-    }
+public extension LDConfig {
+    typealias CacheFactory = (_ cacheKey: String?, _ logger: OSLog) -> KeyedValueCaching
```

Comment:
This was dictated by the `convertCacheData` function, which is migrating UserDefaults with key = nil

Reply:
I think that's actually in line with my later comment about this conversion being something associated with a specific cache implementation. Our existing implementation could take its cache key, but when doing the conversion bit, it can ignore that key then since it knows it didn't use to use it.

Also, at this point, I think we could argue that our cache converter could probably be cleaned up some. I don't think we have to support a v5 -> v9 migration path. So we could probably remove those steps entirely. What do you think @tanderson-ld

Reply:
I agree.  Dropping support for converting v5 to remove the optionality of the cacheKey is a good tradeoff.